GetIntrinsic.js

```diff
@@ -7,7 +7,24 @@
 
 var undefined;
 
+var $Function = Function;
 var $TypeError = TypeError;
+var $SyntaxError = SyntaxError;
+
+var getESNextFunction = function (name, syntax) {
+	try {
+		return $Function("'use strict'; return " + syntax + ';')();
+	} catch (e) {
+		if (!(e instanceof $SyntaxError)) {
+			try {
+				// This will also prevent bundlers from naively including these files in the bundle:
+				// eslint-disable-next-line global-require
+				return require('./GetIntrinsic/' + name + '.js');
```

Comment:
why use `require` at all, rather than just inlining the code into a `Function` call?

Reply:
As a fallback in case `es‑abstract` is used in an environment with [`‑‑disallow‑code‑generation‑from‑strings`](https://nodejs.org/api/cli.html#cli_disallow_code_generation_from_strings).

---

Or a bundled **CommonJS** environment that has **CSP** where the bundler was configured to allow requiring those files but load them through separate `<script>` elements.

Reply:
I'm not concerned with node flags, lots of code breaks when those are used.

Are you saying it's a common/practical approach to automatically factor out CJS modules that use eval so they're in a separate bundle from the rest of app code?